service/accounts/openstack_manager.py

```diff
@@ -536,45 +577,56 @@ def os_delete_account(self, username, projectname):
             self.user_manager.delete_user(username)
         return True
 
-    @classmethod
-    def hashpass(cls, username, strategy=None):
+    def hashpass(self, username, strategy=None):
         """
         Create a unique password using 'Username' as the wored
         and the SECRET_KEY as your salt
         """
+        # Get salt from config or settings
+        try:
+            secret_salt = self.cloud_config['user']['password_salt']
+        except KeyError:
+            logger.warn("Cloud config ['user']['password_salt'] is missing -- using deprecated secrets.SECRET_SEED")
+            secret_salt = SECRET_SEED
+        secret_salt = str(secret_salt).translate(None, string.punctuation)
+
+        # Get strategy from config or settings
         if not strategy:
-            strategy = DEFAULT_PASSWORD_LOOKUP
+            try:
+                strategy = self.cloud_config['user']['password_lookup']
+            except KeyError:
+                logger.warn("Cloud config ['user']['password_lookup'] is missing -- using deprecated settings.DEFAULT_PASSWORD_LOOKUP")
+                strategy = DEFAULT_PASSWORD_LOOKUP
 
         if not strategy\
                 or strategy == 'old_hashpass':
-            return cls.old_hashpass(username)
+            return self.old_hashpass(username, secret_salt)
         if strategy == 'crypt_hashpass':
-            return cls.crypt_hashpass(username)
+            return self.crypt_hashpass(username, secret_salt)
         elif strategy == 'salt_hashpass':
-            return cls.salt_hashpass(username)
+            return self.salt_hashpass(username)
         else:
             raise ValueError(
                 "Invalid DEFAULT_PASSWORD_LOOKUP: %s"
                 % DEFAULT_PASSWORD_LOOKUP)
 
-    @classmethod
-    def old_hashpass(cls, username):
+    def old_hashpass(self, username):
         from hashlib import sha1
         return sha1(username).hexdigest()
 
-    @classmethod
-    def salt_hashpass(cls, username):
+    def salt_hashpass(self, username, secret_salt):
         from hashlib import sha256
-        secret_salt = SECRET_SEED.translate(None, string.punctuation)
+        if not secret_salt:
+            secret_salt = SECRET_SEED.translate(None, string.punctuation)
         password = sha256(secret_salt + username).hexdigest()
```

Comment:
Unless i'm misunderstanding this:
```diff
- password = sha256(secret_salt + username).hexdigest()
+ password = sha256(secret + salt + username).hexdigest()
```

If we're using a `cloud_config` password per provider, the regular `hash_pass` would be:
```
password = sha256(cloud_pass + username).hexdigest()
```

If we want to salt it with our application secret (which we should just always do:
```
password = sha256(cloud_pass + SECRET_SEED + username).hexdigest()
```